src/dex/aave-v3/utils.ts

```diff
@@ -0,0 +1,156 @@
+import { BytesLike } from 'ethers';
+import { defaultAbiCoder, Interface } from '@ethersproject/abi';
+import {
+  extractSuccessAndValue,
+  stringDecode,
+  uint8ToNumber,
+} from '../../lib/decoders';
+import {
+  MultiCallParams,
+  MultiResult,
+  MultiWrapper,
+} from '../../lib/multi-wrapper';
+import Web3 from 'web3';
+import POOL_ABI from '../../abi/AaveV3_lending_pool.json';
+import { AaveToken } from './types';
+
+export const decodeATokenFromReserveData = (
+  result: MultiResult<BytesLike> | BytesLike,
+): string => {
+  const [isSuccess, toDecode] = extractSuccessAndValue(result);
+
+  if (!isSuccess || toDecode === '0x') {
+    return 'error';
+  }
+
+  const decoded = defaultAbiCoder.decode(
+    [
+      'tuple(uint256) configuration',
+      'uint128 liquidityIndex',
+      'uint128 currentLiquidityRate',
+      'uint128 variableBorrowIndex',
+      'uint128 currentVariableBorrowRate',
+      'uint128 currentStableBorrowRate',
+      'uint40 lastUpdateTimestamp',
+      'uint16 id',
+      'address aTokenAddress',
+      'address stableDebtTokenAddress',
+      'address variableDebtTokenAddress',
+      'address interestRateStrategyAddress',
+      'uint128 accruedToTreasury',
+      'uint128 unbacked',
+      'uint128 isolationModeTotalDebt',
+    ],
+    toDecode,
+  );
+
+  return decoded.aTokenAddress;
```

Comment:
I think better to return already `toLowercase()` from here

Reply:
Done.